File: src/main/java/com/travel/global/result/ResultSet.java
package com.travel.global.result;

import lombok.Getter;

import java.util.List;

@Getter
public class ResultSet {

    public static ResultApi resultData(Object object) {
        ResultApi resultApi = new ResultApi();
        resultApi.setObject(ResultType.resultData(object));

        return resultApi;
    }

    public static ResultApi resultList(List<?> list) {
        return resultList(list, null);
    }

    public static ResultApi resultList(List<?> list, Long listCount) {
        ResultApi resultApi = new ResultApi();
        resultApi.setObject(ResultType.resultList(list, listCount));

        return resultApi;
    }


}

File: src/main/kotlin/com/travel/global/result/ResultSet.kt
package com.travel.global.result


object ResultSet {
    fun resultData(obj: Any): ResultApi {
         return ResultApi(ResultType.resultData(obj));
    }

    fun resultList(list: List<*>): ResultApi {
        return ResultApi(ResultType.resultList(list, null))
    }

    @JvmOverloads
    fun resultList(list: List<*>, listCount: Long?): ResultApi {
        return ResultApi(ResultType.resultList(list, listCount))
    }
}

